adafruit_rfm/rfm_common.py

```diff
@@ -0,0 +1,545 @@
+        (limited by chip's FIFO size and appended headers).
+        if the propert radiohead is True then this appends a 4 byte header
+        to be compatible with the RadioHead library.
+        The header defaults to using the initialized attributes:
+        (destination,node,identifier,flags)
+        It may be temporarily overidden via the kwargs - destination,node,identifier,flags.
+        Values passed via kwargs do not alter the attribute settings.
+        The keep_listening argument should be set to True if you want to start listening
+        automatically after the packet is sent. The default setting is False.
+
+        Returns: True if success or False if the send timed out.
+        """
+        # Disable pylint warning to not use length as a check for zero.
+        # This is a puzzling warning as the below code is clearly the most
+        # efficient and proper way to ensure a precondition that the provided
+        # buffer be within an expected range of bounds. Disable this check.
+        # pylint: disable=len-as-condition
+        assert 0 < len(data) <= self.max_packet_length
+        # pylint: enable=len-as-condition
+        self.idle()  # Stop receiving to clear FIFO and keep it clear.
+        # Combine header and data to form payload
+        if self.radiohead:
+            payload = bytearray(4)
+            if destination is None:  # use attribute
+                payload[0] = self.destination
+            else:  # use kwarg
+                payload[0] = destination
+            if node is None:  # use attribute
+                payload[1] = self.node
+            else:  # use kwarg
+                payload[1] = node
+            if identifier is None:  # use attribute
+                payload[2] = self.identifier
+            else:  # use kwarg
+                payload[2] = identifier
+            if flags is None:  # use attribute
+                payload[3] = self.flags
+            else:  # use kwarg
+                payload[3] = flags
+            payload = payload + data
+        elif destination is not None:  # prepend destination for non RH packets
+            payload = destination.to_bytes(1, "big") + data
+        else:
+            payload = data
+        self.fill_fifo(payload)
+        # Turn on transmit mode to send out the packet.
+        self.transmit()
+        # Wait for packet_sent interrupt with explicit polling (not ideal but
+        # best that can be done right now without interrupts).
+        timed_out = await asyncio_check_timeout(
+            self.packet_sent, self.xmit_timeout, self.timeout_poll
+        )
+        # Listen again if necessary and return the result packet.
+        if keep_listening:
+            self.listen()
+        else:
+            # Enter idle mode to stop receiving other packets.
+            self.idle()
+        self.clear_interrupt()
+        return not timed_out
+
+    send = asyncio_to_blocking(asyncio_send)
```

Comment:
Does this inherit the doc string too?

Reply:
Good question? How can I find out? I'm not sure how the documentation will handle this -- will "send" show up a a separate function? Is there a better way to do this?

Reply:
I'm not sure either. Local docs can be built using sphinx: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/sharing-our-docs-on-readthedocs#sphinx-2615853

Reply:
Thanks -- I'll check it out.

Reply:
"send" does not show up in the Sphinx docs.   I can add a docstring, Would something like this be OK? If I just replicate the docstring then the entry  generated by sphinx does not show the argument and keywords

```
   send = asyncio_to_blocking(asyncio_send)
    """Non-asyncio wrapper to Send a string of data using the transmitter 
    using the same arguments and keywords as asyncio_send()
    """
```

This shows up  in the docs as 
```
send(*args, **kwargs)[¶](https://github.com/adafruit/Adafruit_CircuitPython_RFM/pull/1#adafruit_rfm.rfm_common.RFMSPI.send)
Non-asyncio wrapper to Send a string of data using the transmitter using the same arguments and keywords as asyncio_send()
```

Reply:
I made a new commit with some additional docstrings. Let me know what you  think

Reply:
Will look once docs are built. Thanks!